scripts/setup-macos.sh

```diff
@@ -86,23 +90,33 @@ function cmake_install {
   mkdir -p "${BINARY_DIR}"
   cmake -Wno-dev -B"${BINARY_DIR}" \
     -GNinja \
+    -DCMAKE_INSTALL_PREFIX="$INSTALL_DEPS_PREFIX" \
     -DCMAKE_CXX_STANDARD=17 \
-    "${INSTALL_PREFIX+-DCMAKE_PREFIX_PATH=}${INSTALL_PREFIX-}" \
-    "${INSTALL_PREFIX+-DCMAKE_INSTALL_PREFIX=}${INSTALL_PREFIX-}" \
     -DCMAKE_CXX_FLAGS="${COMPILER_FLAGS}" \
     "$@"
   ninja -C "${BINARY_DIR}" install
 }
 
-function update_brew {
-  /usr/local/bin/brew update --force --quiet
+function install_build_prerequisites_brew {
+  for pkg in ${MACOS_DEPS_BREW}
+  do
+      brew install --formula $pkg && echo "Installation of $pkg is successful" || brew upgrade --formula $pkg
+  done
 }
 
-function install_build_prerequisites {
-  for pkg in ${MACOS_DEPS}
+function install_build_prerequisites_macports {
+  for pkg in ${MACOS_DEPS_MACPORTS}
   do
-    brew install --formula $pkg && echo "Installation of $pkg is successful" || brew upgrade --formula $pkg
+    sudo port install $pkg
   done
+}
+
+function install_build_prerequisites {
+  if hash brew 2>/dev/null; then
+    install_build_prerequisites_brew
+  else
```

Comment:
Check for `port`; if neither `brew` or `port` is available print an error message asking the user to install one.